contracts/zap/decollateralize/SolidZapDecollateralize.sol

```diff
@@ -3,12 +3,108 @@ pragma solidity 0.8.18;
 
 import "./BaseSolidZapDecollateralize.sol";
 
+interface SWManager {
+    function bulkDecollateralizeTokens(
+        uint[] calldata batchIds,
+        uint[] calldata amountsIn,
+        uint[] calldata amountsOutMin
+    ) external;
+
+    function getBatchCategory(uint batchId) external view returns (uint);
+}
+
 /// @author Solid World
 contract SolidZapDecollateralize is BaseSolidZapDecollateralize {
+    using GPv2SafeERC20 for IERC20;
+
     constructor(
         address _router,
         address _weth,
         address _swManager,
         address _forwardContractBatch
     ) BaseSolidZapDecollateralize(_router, _weth, _swManager, _forwardContractBatch) {}
+
+    /// @inheritdoc ISolidZapDecollateralize
+    function zapDecollateralize(
+        address inputToken,
+        uint inputAmount,
+        address crispToken,
+        bytes calldata swap,
+        address dustReceiver,
+        DecollateralizeParams calldata decollateralizeParams
+    ) external nonReentrant {
+        _prepareToSwap(inputToken, inputAmount);
+        _zapDecollateralize(
+            inputToken,
+            inputAmount,
+            crispToken,
+            swap,
+            dustReceiver,
+            decollateralizeParams,
+            msg.sender
+        );
+    }
+
+    /// @inheritdoc ISolidZapDecollateralize
+    function zapDecollateralize(
+        address inputToken,
+        uint inputAmount,
+        address crispToken,
+        bytes calldata swap,
+        address dustReceiver,
+        DecollateralizeParams calldata decollateralizeParams,
+        address recipient
+    ) external nonReentrant {
+        _prepareToSwap(inputToken, inputAmount);
+        _zapDecollateralize(
+            inputToken,
+            inputAmount,
+            crispToken,
+            swap,
+            dustReceiver,
+            decollateralizeParams,
+            recipient
+        );
+    }
+
+    function _zapDecollateralize(
+        address inputToken,
+        uint inputAmount,
+        address crispToken,
+        bytes calldata swap,
+        address dustReceiver,
+        DecollateralizeParams calldata decollateralizeParams,
+        address recipient
```

Comment:
I think need to add some sanity check for recipient.

Reply:
why bother? if there's anything wrong with the recipient address the tx will revert anyways. There's not really any risk to mitigate with the recipient, if it's a wrong address, or 0x0 or the zap itself, funds will just be lost.

This version with recipient parameter is an extension so other contracts can integrate with the zap, so we can assume they test the integration and provide correct inputs to the function, or risk losing funds

Reply:
Okay, if you think so.